src/app/app-platform/ContentAppPlatform.cpp

```diff
@@ -482,12 +537,30 @@ CHIP_ERROR ContentAppPlatform::ManageClientAccess(Messaging::ExchangeManager & e
 
     Access::Privilege vendorPrivilege = mContentAppFactory->GetVendorPrivilege(targetVendorId);
 
+    NodeId subjectNodeId    = sessionHandle->GetPeer().GetNodeId();
+    FabricIndex fabricIndex = sessionHandle->GetFabricIndex();
+
+    // first, delete existing ACLs for this nodeId
```

Comment:
This runtime mutates the ACL. This seems to fall outside the scope of this PR about "opcert callbacks" and 'recommissioning"

Also, this is a node deciding to grant privileges on its own without an existing admininistrator's knowledge/control. Can you explain how this came to be?

Reply:
This is the app platform code using for casting and the ACL entries created follow the device library spec (casting video player section). The ContentAppPlatform is re-creating the ACL entry for the node after re-commissioning it. This ensures the ACL entry is up-to-date since its possible that new apps have been loaded which grant access to the given nodeId.

Reply:
This code is very opaque. It may well be spec compliant, but I read the cluster spec and there are many statements like `The method for conveying this information between the Content App and the Casting Video Player is specific to the vendor of the Casting Video Player`. I think it woould help everyone if much more details about the rules applied here were explicitly documented in the code. Otherwise it's very possible that authors other than yourself or close colleagues would be able to maintain this code.

Reply:
added more comments in header file and in the implementation body of what is happening